examples/jsm/exporters/USDZExporter.js

```diff
@@ -331,9 +331,56 @@ function buildMaterial( material ) {
 	const pad = '            ';
 	const parameters = [];
 
+	const texturesTransforms = [];
+	const textures = [];
+	let texture;
+
+	function buildColorWithAlpha( color ) {
+
+		return `(${ color.r }, ${ color.g }, ${ color.b }, 1)`;
+	
+	}
+
+	function prepareTextureTransform(texture) {
+		return `
+		def Shader "Transform2D_${ texture.id }" (
+			sdrMetadata = {
+				string role = "math"
+			}
+		)
+		{
+			uniform token info:id = "UsdTransform2d"
+			float2 inputs:in.connect = </Materials/Material_${ material.id }/uvReader_st.outputs:result>
+			float2 inputs:scale = (${ texture.repeat.x },${ texture.repeat.y })
+			float2 inputs:translation = (${ texture.offset.x },${ texture.offset.y })
+			float2 outputs:result
+		}
+		`;
+	}
+
+
+
 	if ( material.map !== null ) {
 
-		parameters.push( `${ pad }color3f inputs:diffuseColor.connect = </Textures/Texture_${ material.map.id }.outputs:rgb>` );
+		parameters.push( `${ pad }color3f inputs:diffuseColor.connect = </Materials/Material_${ material.id }/Texture_${ material.map.id }.outputs:rgb>` );
+
+		texture = material.map;
+		texturesTransforms.push(prepareTextureTransform(texture));
+		textures.push(`
+
+		def Shader "Texture_${ texture.id }"
+		{
+			uniform token info:id = "UsdUVTexture"
+			asset inputs:file = @textures/Texture_${ texture.id }.jpg@
+			float2 inputs:st.connect = </Materials/Material_${ material.id }/Transform2D_${ texture.id }.outputs:result>
+			float4 inputs:scale = (1, 1, 1, 1)
+			float4 inputs:bias = (${material.color.r}, ${material.color.g}, ${material.color.b}, 0)
```

Comment:
this should theoretically add material color to rgb components of the diffuse map, but it does not work! Playing with scale I can achieve some coloration to happen, but bias does nothing.